app/mailboxes/inbox_mailbox.rb

```diff
@@ -0,0 +1,64 @@
+# typed: false
+
+class InboxMailbox < ApplicationMailbox
+  before_processing :required_info
+
+  def process
+    c = Comment.new(user: sending_user, comment: mail.decoded, is_from_email: true)
+    if parent.is_a?(Comment)
+      c.story_id = parent.story_id
+      c.parent_comment_id = parent.id
+    else
+      c.story_id = parent.id
+    end
+
+    # TODO: Figure out what to do here.
+    if c.save
+      Rails.logger.info("comment saved")
+    else
+      Rails.logger.info("comment NOT saved")
+    end
+  end
+
+  private
+
+  def required_info
+    if mail.decoded == "" || sending_user.nil? || parent.nil?
+      # We could email the user about the bounce, but that doesn't seem
+      # to align with current behaviour.
+      bounced!
+    end
+  end
+
+  def parent
+    return @parent if @parent
+
+    # Emails In-Reply-To formatted like
+    # story.<short_id>.<timestamp of story creation>@<app domain>
+    # We throw away the timestamp, since we only care about the short ID.
+    irt = mail.in_reply_to.to_s.gsub(/[^A-Za-z0-9@\.]/, "")
+
+    if (m = irt.match(/^comment\.([^\.]+)\.\d+@/))
+      @parent = Comment.where(short_id: m[1]).first
+    elsif (m = irt.match(/^story\.([^\.]+)\.\d+@/))
+      @parent = Story.where(short_id: m[1]).first
+    end
+
+    @parent
+  end
+
+  def user_token
+    # RoR "to" gives us an array. Joining here might not be the best?
+    mail.to.join(" ").to_s.gsub(/^#{Rails.application.shortname}-/, "").gsub(/@.*/, "")
```

Comment:
```suggestion
    mail.to.join(" ").to_s.gsub(/^#{Rails.application.shortname}-/, "").gsub(/@.*/, "")
    mail.to.select {|e| e =~ /^#{Rails.application-shortname}\-/ }[/\-([^@]*)@/, 1].to_s
```

Not sure about the `\` on the hyphen though.

Reply:
This code change is so you can rm the comment. Your regexp didn't quite work; the ^ would match start-of-line so joining the `to` addresses with a space meant it would only look for a token in the first email in the `To` line. I tweaked it to be a little more explicit that we're trying to find an address that looks like our reply-to address and then pull the token out of it.